src/common/node/fileSystem.ts

```diff
@@ -1,28 +1,33 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for details.
 
-import * as fs from "fs";
+import * as nodeFs from "fs";
 import * as path from "path";
 import * as Q from "q";
 
 export class FileSystem {
+    private fs: typeof nodeFs;
```

Comment:
I think using a naming convention on member vars would be better, e.g. m_ or just _, to avoid naming conflicts/confusion like you have with fs